ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TopicInfoCache.java

```diff
@@ -118,7 +119,21 @@ public TopicInfoCache(
       return Optional.empty();
     }
 
-    return Optional.of(cache.getUnchecked(topicName));
+    final TopicInfo topicInfo = cache.getUnchecked(topicName);
+
+    // The problem I am fixing is that sometimes the first record found in a topic has a null value.
+    // This causes the QuerySchemas#getTopicInfo() to return a EMPTY_SCHEMA for the value.
+    // When that info is obtained from the load function, then it is kept in the cache for some
+    // time. I assume that the null value is just something that may not
```

Comment:
Thanks for the detailed comment. I'm still a bit confused about what problem this is solving, though. Is there a race condition of some sort? Is the issue nondeterministic or would tests fail deterministically if we didn't refresh the cache in this way?

Reply:
I just removed it. Seems there's no issue with it. I tested it and it worked without it.